cpp/src/slice2java/Gen.cpp

```diff
@@ -3049,12 +3049,26 @@ Slice::Gen::TypesVisitor::visitEnum(const EnumPtr& p)
     }
     out << ';';
 
-    out << sp << nl << "public int value()";
+    out << sp;
+    out << nl << "/**";
+    out << nl << " * Returns the integer value of this enumerator.";
+    out << nl << " *";
+    out << nl << " * @return the integer value of this enumerator";
+    out << nl << " */";
+    out << nl << "public int value()";
     out << sb;
     out << nl << "return _value;";
     out << eb;
 
-    out << sp << nl << "public static " << name << " valueOf(int v)";
+    out << sp;
+    out << nl << "/**";
+    out << nl << " * Returns the {@code " << name << "} type corresponding to the given integer value.";
```

Comment:
I find type here confusing, there is a single type. I think using `instance` instead of type would be clear.

```
Returns the {@code " << name << "} instance corresponding...
```